inttest/common/footloosesuite.go

```diff
@@ -189,6 +189,13 @@ func (s *FootlooseSuite) keepEnvironment() bool {
 	}
 }
 
+func k0sConfigFlag() string {
+	if os.Getenv("K0S_CONFIG") != "" {
+		return "--config /etc/k0s.conf"
```

Comment:
Is this supposed to always return the hardcoded path? The code below will then have the same source and destination:

```go
		volumes = append(volumes, config.Volume{
			Type:        "bind",
			Source:      configFile, // <= will be /etc/k0s.conf when K0S_CONFIG is set.
			Destination: "/etc/k0s.conf",
		})
```

Reply:
No, the `Source` will be `osGetenv("K0S_CONFIG")`, which is a path on the host running the test. This will be bind mounted to `/etc/k0s.conf` in the footloose container, and `k0sConfigFlag()` will append the `--config /etc/k0s.conf` to `k0s` if the bindmount was added to footloose regardless of what the env var was set to. (i.e the test runs with a specified config instead of the defaults).

```diff
@@ -402,6 +410,15 @@ func (s *FootlooseSuite) createConfig() config.Config {
 
        volumes = append(volumes, s.ExtraVolumes...)
 
+       configFile := os.Getenv("K0S_CONFIG")
+       if configFile != "" {
+               volumes = append(volumes, config.Volume{
+                       Type:        "bind",
+                       Source:      configFile,
+                       Destination: "/etc/k0s.conf",
+               })
+       }
+
        portMaps := []config.PortMapping{
                {
                        ContainerPort: 22,

```